### ISSUE TITLE
Bookmarklet is not draggable

### ISSUE THREAD

**Issue by perryprog** [adzerk-oss/boot-bookmarklet#2]
After running the command `boot build` with a configuration like so:
```clojure
(set-env! :dependencies '[[adzerk/boot-bookmarklet "0.2.0" :scope "test"]])
(require '[adzerk.boot-bookmarklet :refer (bookmarklet external-bookmarklet)])


(deftask build
	[]
	(comp
		(speak)
		(bookmarklet :ids #{"hello"})
		(target)))
```

I get the output
```
Writing main.cljs.edn...
Compiling ClojureScript...
WARNING: No ClojureScript in project dependencies but ClojureScript was found in classpath. Adding direct dependency is adviced.
• main.js
Writing bookmarklets.html...
Writing target dir(s)...
```

Then running `open target/bookmarklets.html` will open it in a browser that will look like [so](https://gist.github.com/anonymous/9fe6eb6c5e69f26403600ca1ad9721fb). Clicking this will produce the expected result, But I can't drag the link. The URL of the `<a>` tag is:
```
javascript:%20var%20CLOSURE_UNCOMPILED_DEFINES%20%3D%20null%3B%0Aif%28typeof%20goog%20%3D%3D%20%22undefined%22%29%20document.write%28%27%3Cscript%20src%3D%22main.out%2Fgoog%2Fbase.js%22%3E%3C%2Fscript%3E%27%29%3B%0Adocument.write%28%27%3Cscript%20src%3D%22main.out%2Fcljs_deps.js%22%3E%3C%2Fscript%3E%27%29%3B%0Adocument.write%28%27%3Cscript%3Eif%20%28typeof%20goog%20%21%3D%20%22undefined%22%29%20%7B%20goog.require%28%22boot.cljs.main647%22%29%3B%20%7D%20else%20%7B%20console.warn%28%22ClojureScript%20could%20not%20load%20%3Amain%2C%20did%20you%20forget%20to%20specify%20%3Aasset-path%3F%22%29%3B%20%7D%3B%3C%2Fscript%3E%27%29%3B%0A
```

That URL would be the following Javascript:
```javascript
 var CLOSURE_UNCOMPILED_DEFINES = null;
if(typeof goog == "undefined") document.write('<script src="main.out/goog/base.js"></script>');
document.write('<script src="main.out/cljs_deps.js"></script>');
document.write('<script>if (typeof goog != "undefined") { goog.require("boot.cljs.main647"); } else { console.warn("ClojureScript could not load :main, did you forget to specify :asset-path?"); };</script>');
```

It looks like it's trying to load a local file, which does not seem right for a bookmarklet. It's also impossible to drag the URL.

You can view all the files [here](https://github.com/perryprog/book-test/tree/master).

**Comment by perryprog** [comment]
Side note: I'm new to boot and ClojureScript, so any help is appreciated.

**Comment by daveyarwood** [comment]
Hello, and welcome to ClojureScript! :)

What's happening here is that the ClojureScript compilation step isn't finding any cljs source files to compile into JavaScript, because Boot isn't aware of any source paths by default.

You can fix that by specifying in your `build.boot` where to find the source files:

```clojure
(set-env!
  :source-paths #{"src"}
  :dependencies '[[adzerk/boot-bookmarklet "0.2.0" :scope "test"]])
```

**Comment by perryprog** [comment]
Ah 👍 

Thanks for the quick response!

**Comment by perryprog** [comment]
I'm still having trouble dragging the URL.
It contains the following javascript:
```javascript
 if(typeof Math.imul == "undefined" || (Math.imul(0xffffffff,5) == 0)) {
    Math.imul = function (a, b) {
        var ah  = (a >>> 16) & 0xffff;
        var al = a & 0xffff;
        var bh  = (b >>> 16) & 0xffff;
        var bl = b & 0xffff;
        // the shift by 0 fixes the sign on the high part
        // the final |0 converts the unsigned value into a signed value
        return ((al * bl) + (((ah * bl + al * bh) << 16) >>> 0)|0);
    }
}


if("undefined"===typeof b)var b=function(){throw Error("No *print-fn* fn set for evaluation environment");};if("undefined"===typeof c)var c=function(){throw Error("No *print-err-fn* fn set for evaluation environment");};if("undefined"===typeof g)var g=null;if("undefined"!==typeof Symbol){var h=Symbol;"object"!=typeof h||!h||h instanceof Array||h instanceof Object||Object.prototype.toString.call(h)}
var k="undefined"!==typeof Math.imul&&0!==Math.imul(4294967295,5)?function(a,d){return Math.imul(a,d)}:function(a,d){var e=a&65535,f=d&65535;return e*f+((a>>>16&65535)*f+e*(d>>>16&65535)<<16>>>0)|0};function l(a){a=k(a|0,-862048943);a=0^(k(a<<15|a>>>-15,461845907)|0);a=(k(a<<13|a>>>-13,5)+-430675100|0)^0;a=k(a^a>>>16,-2048144789);k(a^a>>>13,-1028477387)}l(1);l(0);alert("Hello, World");
```

**Comment by daveyarwood** [comment]
Not being able to drag the URL typically means that the URL is too long, i.e. there is too much generated JavaScript code.

In this case, though, I think the URL should be just small enough to work. In fact, I used boot-bookmarklet to generate this external bookmarklet link:

```
javascript: %28function%20%28%29%20%7B%20var%20jsCode%20%3D%20document.createElement%28%27script%27%29%3B%20jsCode.setAttribute%28%27src%27%2C%20%27https%3A%2F%2Fdl.dropboxusercontent.com%2Fu%2F100360569%2Fbagel_factory.js%27%29%3B%20document.body.appendChild%28jsCode%29%3B%20%7D%28%29%29%3B
```

which is only 262 characters, and I am not able to drag it either. This was working ~7-10 months ago when I was last testing boot-bookmarklet, and nothing has changed since then, which leads me to suspect that there was a change to the Chrome browser since then. I tried the same link in Firefox and it worked.

Your link works in Firefox (and not Chrome) as well.

WTF, Chrome? This is disappointing.

**Comment by perryprog** [comment]
I'm using Safari, but Chrome works fine. Clicking it works strangely.

**Comment by perryprog** [comment]
Yeah, it works in Firefox for me too. 😖 

I also feel like the link the boot-bookmarklet generators should only be `alert('foo')` and not all of that other code. Why is that stuff there?

**Comment by daveyarwood** [comment]
I am at least able to manually create bookmarklets in my bookmarks bar by doing this, as a workaround:

- Right-click the bookmarklet link ->Copy Link Address.
- Right-click the bookmarks bar -> Add Page...
- Change the title to whatever you'd like.
- Replace the URL with the one you have copied.

**Comment by perryprog** [comment]
Tested some other online bookmarklets, they work fine. This makes me think it's a bug in boot-bookmarklet.

**Comment by daveyarwood** [comment]
OK, that's encouraging that it's something we could potentially fix.

Do you have an example of a bookmarklet link that is draggable, for comparison?

**Comment by perryprog** [comment]
Sure. 
```
javascript:function%20iptxt(){var%20d=document;try{if(!d.body)throw(0);window.location='http://www.instapaper.com/text?u='+encodeURIComponent(d.location.href);}catch(e){alert('Please%20wait%20until%20the%20page%20has%20loaded.');}}iptxt();void(0) 
``` is from Instapaper.

Working link [here](https://www.instapaper.com/save).

**Comment by daveyarwood** [comment]
Oh, I think figured it out! If you inspect the HTML generated by boot-bookmarklet, you'll see that each link is an `a` containing an `h1`. Apparently this causes some browsers not to make the link draggable for some reason.

I stuck the following in an HTML file:

```html
<html>
  <head>
    <title>Bookmarklets</title>
  </head>
  <body>
    <div>
      <h1>
        <a href="javascript:alert('hello!');">
          hello.js
        </a>
      </h1>

      <a href="javascript:alert('hello?');">
        <h1>
          broken.js
        </h1>
      </a>
    </div>
  </body>
</html>
```

In Firefox, I am able to drag both links.

In Chrome and Safari, I am able to drag the first one, but not the second one.

**Comment by perryprog** [comment]
Cool! Could you implement this?

**Comment by perryprog** [comment]
Thanks so much!

**Comment by daveyarwood** [comment]
Fixed in boot-bookmarklet 0.2.1.

Thanks for pointing this out, I don't think I would have noticed!

**Comment by perryprog** [comment]
Yeah, it's working perfectly now. :birthday: for all.
